protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+impl BridgeContractCounterparty for MovementClient {
+	type Address = MovementAddress;
+	type Hash = [u8; 32];
+
+	async fn lock_bridge_transfer_assets(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		hash_lock: HashLock<Self::Hash>,
+		time_lock: TimeLock,
+		initiator: InitiatorAddress<Vec<u8>>,
+		recipient: RecipientAddress<Self::Address>,
+		amount: Amount,
+	) -> BridgeContractCounterpartyResult<()> {
+		//@TODO properly return an error instead of unwrapping
+		let args = vec![
+			self.to_bcs_bytes(&initiator.0).unwrap(),
+			self.to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+			self.to_bcs_bytes(&hash_lock.0).unwrap(),
+			self.to_bcs_bytes(&time_lock.0).unwrap(),
+			self.to_bcs_bytes(&recipient.0).unwrap(),
+			self.to_bcs_bytes(&amount.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"lock_bridge_transfer_assets",
+			self.counterparty_type_args(Call::Lock),
+			args,
+		);
+		let _ = utils::send_aptos_transaction(&self.rest_client, &mut self.signer, payload)
+			.await
+			.map_err(|_| BridgeContractCounterpartyError::LockTransferAssetsError);
+		Ok(())
+	}
+
+	async fn complete_bridge_transfer(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+		preimage: HashLockPreImage,
+	) -> BridgeContractCounterpartyResult<()> {
+		let args = vec![
+			self.to_bcs_bytes(&self.signer.address()).unwrap(),
+			self.to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+			self.to_bcs_bytes(&preimage.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"complete_bridge_transfer",
+			self.counterparty_type_args(Call::Complete),
+			args,
+		);
+		let _ = utils::send_aptos_transaction(&self.rest_client, &mut self.signer, payload)
+			.await
+			.map_err(|_| BridgeContractCounterpartyError::CompleteTransferError);
+		Ok(())
+	}
+
+	async fn abort_bridge_transfer(
+		&mut self,
+		bridge_transfer_id: BridgeTransferId<Self::Hash>,
+	) -> BridgeContractCounterpartyResult<()> {
+		let args = vec![
+			self.to_bcs_bytes(&self.signer.address()).unwrap(),
+			self.to_bcs_bytes(&bridge_transfer_id.0).unwrap(),
+		];
+		let payload = utils::make_aptos_payload(
+			self.counterparty_address,
+			COUNTERPARTY_MODULE_NAME,
+			"abort_bridge_transfer",
+			self.counterparty_type_args(Call::Abort),
+			args,
+		);
+		let _ = utils::send_aptos_transaction(&self.rest_client, &mut self.signer, payload)
+			.await
+			.map_err(|_| BridgeContractCounterpartyError::AbortTransferError);
+		Ok(())
+	}
+
+	async fn get_bridge_transfer_details(
+		&mut self,
+		_bridge_transfer_id: BridgeTransferId<Self::Hash>,
+	) -> BridgeContractCounterpartyResult<Option<BridgeTransferDetails<Self::Hash, Self::Address>>>
+	{
+		// let _ = utils::send_view_request(
+		// 	self.rest_client,
+		// 	self.counterparty_address,
+		// 	"atomic_bridge_counterparty".to_string(),
+		// );
+		todo!();
```

Comment:
Needs to be implemented for the client implementation to work

Reply:
Yes.